lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1411,49 +1580,47 @@ public function __toString() {
 	 * @return string The processed HTML.
 	 */
 	public function get_updated_html() {
-		// Short-circuit if there are no updates to apply.
-		if ( ! count( $this->classname_updates ) && ! count( $this->attribute_updates ) ) {
+		if ( count( $this->classname_updates ) || count( $this->attribute_updates ) ) {
+			$this->apply_html_updates_and_stay_on_current_tag();
+
+			return $this->html;
+		} else {
 			return $this->updated_html . substr( $this->html, $this->updated_bytes );
 		}
+	}
 
-		/*
-		 * Parsing is in progress – let's apply the attribute updates without moving on to the next tag.
-		 *
-		 * In practice:
-		 * 1. Apply the attributes updates to the original HTML
-		 * 2. Replace the original HTML with the updated HTML
-		 * 3. Point this tag processor to the current tag name's end in that updated HTML
-		 */
+	/**
+	 * Applies all the enqueued attribute updates without advancing any
+	 * internal pointers to the next tag.
+	 *
+	 * @since 6.2.0
+	 */
+	private function apply_html_updates_and_stay_on_current_tag() {
```

Comment:
I'd rather emphasise the "replace `$this->html` with `$this->updated_html` part" rather than the `stay_on_current_tag` part. Any ideas for a better name?